packages/@aws-cdk/aws-eks-next/lib/cluster.ts

```diff
@@ -29,47 +29,53 @@ const DEFAULT_CAPACITY_TYPE = ec2.InstanceType.of(ec2.InstanceClass.M5, ec2.Inst
  * An EKS cluster
  */
 export interface ICluster extends IResource, ec2.IConnectable {
-  /**
-   * The VPC in which this Cluster was created
-   */
-  readonly vpc: ec2.IVpc;
 
   /**
    * The physical name of the Cluster
+   *
    * @attribute
    */
   readonly clusterName: string;
 
   /**
-   * The unique ARN assigned to the service by AWS
-   * in the form of arn:aws:eks:
+   * The unique ARN assigned to the service by AWS in the form of arn:aws:eks:
+   *
    * @attribute
    */
   readonly clusterArn: string;
 
   /**
-   * The API Server endpoint URL
+   * The VPC in which this Cluster was created.
+   */
+  readonly vpc?: ec2.IVpc;
```

Comment:
How come these are optional? A cluster is always within a vpc. If you are referring to imported clusters, it should be a runtime error and not a compilation error.